test/module/irohad/multi_sig_transactions/transport_test.cpp

```diff
@@ -51,31 +49,31 @@ TEST(TransportTest, SendAndReceive) {
       .WillByDefault(
           InvokeWithoutArgs(&cv, &std::condition_variable::notify_one));
 
-  std::shared_ptr<shared_model::interface::Peer> peer =
-      makePeer("localhost:50051", "abcdabcdabcdabcdabcdabcdabcdabcd");
-
   auto state = iroha::MstState::empty();
   state += makeTx(1, iroha::time::now(), makeKey(), 3);
   state += makeTx(1, iroha::time::now(), makeKey(), 4);
   state += makeTx(1, iroha::time::now(), makeKey(), 5);
   state += makeTx(1, iroha::time::now(), makeKey(), 5);
 
-  // we want to ensure that server side will call onNewState()
-  // with same parameters as on the client side
-  EXPECT_CALL(*notifications, onNewState(_, state))
-      .WillOnce(Invoke([&peer](auto &p, auto) { EXPECT_EQ(*p, *peer); }));
-
   std::unique_ptr<grpc::Server> server;
 
   grpc::ServerBuilder builder;
   int port = 0;
+  std::string addr = "localhost:";
   builder.AddListeningPort(
-      peer->address(), grpc::InsecureServerCredentials(), &port);
+      addr + "0", grpc::InsecureServerCredentials(), &port);
   builder.RegisterService(transport.get());
   server = builder.BuildAndStart();
   ASSERT_TRUE(server);
   ASSERT_NE(port, 0);
 
+  std::shared_ptr<shared_model::interface::Peer> peer =
```

Comment:
Do we really need `shared_ptr` for that?

Reply:
Not really, but there's a helper function that return shared_ptr